### PR TITLE
Users can only vote once on a review. 

### PR DESCRIPTION
**Pull request by ccjeane**
- Added voters set in comment.java to keep track of who has voted on what comment
- Stores review + voter in a database to keep long-term track of who has voted on what! 
- PlaceReviews keeps track of who is currently logged in so that javascript can access that data during voting.

### CODE REVIEW

File: src/main/java/com/google/sps/data/Comment.java [hunk 1]
```diff
@@ -53,6 +52,12 @@ public Comment(String author, String message, Date timestamp, Long pos, Long neg
     this.timestamp = timestamp;
     this.positive = pos;
     this.negative = neg;
+    this.voters = new HashSet<>();
```
Comment (dterr) [inline src/main/java/com/google/sps/data/Comment.java:55]: You know I love to see the use of a HashSet

File: src/main/webapp/index-script.js [hunk 2]
```diff
@@ -391,14 +390,14 @@ function displayReviewModal() {
 function fetchReviews(placeID) {
   const request = '/review?place_id=' + placeID;
   fetch(request).then(response => response.json()).then((place) => {
-    populateReviews(place.reviews, placeID);
+    populateReviews(place.reviews, placeID, place.currUser);
```
Comment (dterr) [inline src/main/webapp/index-script.js:393]: Warning: Possible merge conflict. No biggie tho

File: src/main/webapp/index-script.js [hunk 3]
```diff
@@ -462,19 +461,23 @@ function generateReview(review) {
...
     }
   });
 
   const downvoteButton = document.createElement('button');
   downvoteButton.innerHTML += '&#128078;' + review.negative;
   downvoteButton.id += "down" + review.id;
   downvoteButton.addEventListener("click", () => {
-    if(upvoteButton.innerHTML[0] != " ") {
+    // Users must be logged in and can only vote once. 
+    if(currUser != null && !review.voters.includes(currUser)) {
```
Comment (dterr) [inline src/main/webapp/index-script.js:477]: ### nit
whitespace before (
```
if ( 
```

File: src/main/java/com/google/sps/data/Comment.java [hunk 4]
```diff
@@ -53,6 +52,12 @@ public Comment(String author, String message, Date timestamp, Long pos, Long neg
...
+  }
+
+  // Add a voter to the voters set.
```
Comment (kcirerick) [inline src/main/java/com/google/sps/data/Comment.java:58]: /** Comments for function headers

File: src/main/java/com/google/sps/data/PlaceReviews.java [hunk 5]
```diff
@@ -38,6 +38,11 @@ public PlaceReviews(String place_id, Comment firstReview, double initialRating)
     this.place_id = place_id;
     this.reviews.add(firstReview);
     this.rating = initialRating;
+    this.reviewers = new HashSet<>();
+  }
+
+  public void setCurrentUser(String person){
```
Comment (kcirerick) [inline src/main/java/com/google/sps/data/PlaceReviews.java:44]: add function header comment

File: src/main/java/com/google/sps/data/PlaceReviews.java [hunk 6]
```diff
@@ -93,13 +98,16 @@ public void addRating(double rating) {
     }
   }
 
+  public void addReviewer(String author){
```
Comment (kcirerick) [inline src/main/java/com/google/sps/data/PlaceReviews.java:101]: comment

File: src/main/webapp/index-script.js (hunk 2 above)
Comment (ccjeane) [reply src/main/webapp/index-script.js:412]: We will fix it once it arises :)

**Pull request merged**